pkg/controller/aws/identity/iamrolepolicyattachment.go

```diff
@@ -0,0 +1,209 @@
+	}
+
+	var attachedPolicyObject *awsiam.AttachedPolicy
+	for i := 0; i < len(observed.AttachedPolicies); i++ {
+		if cr.Spec.PolicyArn == *observed.AttachedPolicies[i].PolicyArn {
+			attachedPolicyObject = &(observed.AttachedPolicies[i])
+			break
+		}
+	}
+
+	if attachedPolicyObject == nil {
+		return resource.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	cr.Status.IAMRolePolicyAttachmentExternalState = cr.BuildExternalStateFromObservation(*attachedPolicyObject)
+
+	return resource.ExternalObservation{
+		ResourceExists: true,
+	}, nil
+}
+
+func (e *rpaExternal) Create(ctx context.Context, mgd resource.Managed) (resource.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.IAMRolePolicyAttachment)
+	if !ok {
+		return resource.ExternalCreation{}, errors.New(errObjectNotIAMRolePolicyAttachment)
+	}
+
+	cr.SetConditions(runtimev1alpha1.Available())
+
+	req := e.client.AttachRolePolicyRequest(&awsiam.AttachRolePolicyInput{
+		PolicyArn: aws.String(cr.Spec.PolicyArn),
+		RoleName:  aws.String(cr.Spec.RoleName),
+	})
+	req.SetContext(ctx)
+
+	_, err := req.Send()
+
+	return resource.ExternalCreation{}, errors.Wrapf(err, "failed to attach the policy with arn %v to role %v", cr.Spec.PolicyArn, cr.Spec.RoleName)
+}
+
+func (e *rpaExternal) Update(ctx context.Context, mgd resource.Managed) (resource.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha1.IAMRolePolicyAttachment)
+	if !ok {
+		return resource.ExternalUpdate{}, errors.New(errObjectNotIAMRolePolicyAttachment)
+	}
+
+	// TODO(soorena776): add more sophisticated Update logice, once we
+	// categorize immutable vs mutable fields (see #727)
+
+	// there is not a dedicated update method, so a basic update is implemented below
+	// based on changes on PolicyArn:
+	// if the previously attached policy is different than what is stated in the spec,
+	// for update it needs to first detaching the previous one, and then attaching the updated one
+	if cr.Status.AttachedPolicyArn == "" || cr.Spec.PolicyArn == cr.Status.AttachedPolicyArn {
+		// update is only necessary if the PolicyArn in the Status is set and is different
+		// from the one in Spec
+		return resource.ExternalUpdate{}, nil
+	}
+
+	dReq := e.client.DetachRolePolicyRequest(&awsiam.DetachRolePolicyInput{
+		PolicyArn: aws.String(cr.Status.AttachedPolicyArn),
+		RoleName:  aws.String(cr.Spec.RoleName),
+	})
+	dReq.SetContext(ctx)
+
+	if _, err := dReq.Send(); err != nil {
+		return resource.ExternalUpdate{}, errors.Wrapf(err, "failed to detach the policy with arn %v to role %v", cr.Status.AttachedPolicyArn, cr.Spec.RoleName)
+	}
+
+	aReq := e.client.AttachRolePolicyRequest(&awsiam.AttachRolePolicyInput{
+		PolicyArn: aws.String(cr.Spec.PolicyArn),
+		RoleName:  aws.String(cr.Spec.RoleName),
+	})
+	aReq.SetContext(ctx)
+	_, err := aReq.Send()
```

Comment:
Will this attach then detach process result in a brief window during which there are no policies attached? Could that affect the resource's functionality? Would it be possible instead to attach the new role, then detach the old one? This would also protect against the case in which the detach succeeds, but the attach fails.